solr/core/src/java/org/apache/solr/handler/component/QueryComponent.java

```diff
@@ -169,6 +160,20 @@ public void prepare(ResponseBuilder rb) throws IOException {
 
     try {
       QParser parser = QParser.getParser(rb.getQueryString(), defType, req);
+
+      SortSpec sortSpec = parser.getSortSpec(true);
+      rb.setSortSpec(sortSpec);
+      boolean someDocs = sortSpec == null || sortSpec.getCount() != 0;
+
+      // Set field flags
```

Comment:
The try-catch is about query parsing / syntax errors; sortSpec & returnFields is separate; perhaps should go after.

Reply:
I was anticipating seeing a rows check in changes to QueryComponent but don't see it yet

Reply:
`sortSpec.getCount()` is the rows for the query. SortSpec encapsulates the pulling out of sort details and rows for the query. The parser fills in SortSpec so wanted to reuse that logic instead of explicitly trying to parse rows myself.

I have a separate commit coming soon once tests pass to actually make SortSpec and SolrReturnFields a bit smarter about rows=0. 

There is some weird intermingling of `GET_SCORES` where it only applies to the `fl` but not to the sort. So if `score` is in the `sort` (which it is by default) then you still calculate scores in a lot of cases.

Reply:
Changed the approach in 8f7cb64 to no couple SortSpec into SolrReturnFields